src/phoenix/experimental/evals/functions/processing.py

```diff
@@ -0,0 +1,51 @@
+from typing import List
+
+from ..models import BaseEvalModel
+
+
+def truncate_text_by_model(model: BaseEvalModel, text: str, token_buffer: int = 0) -> str:
+    max_token_count = model.max_context_size - token_buffer
+    if model.get_token_count_from_text(text) > max_token_count:
+        return (
+            model.get_text_from_tokens(model.get_tokens_from_text(text)[:max_token_count]) + "..."
+        )
+    return text
```

Comment:
```suggestion
    tokens = model.get_tokens_from_text(text)
    if len(tokens) > max_token_count:
        return model.get_text_from_tokens(tokens[:max_token_count) + "..."
    return text
```

Suggestion: We can invoke the tokenizer only once and remove the `get_token_count_from_text` method.

Reply:
Nice, that should've been obvious. Thanks!